project/CompilerSettings.scala

```diff
@@ -19,7 +19,9 @@ object CompilerSettings {
     Seq(
       "-Xfuture",
       "-Ypartial-unification",
-      "-Ywarn-dead-code",
+      // To prevent "dead code following this construct" error when using * mockito-scala library
+      // https://github.com/mockito/mockito-scala/issues/29
+      // "-Ywarn-dead-code",
```

Comment:
I don't like that we lose this checking in the main code if we can disable it only for tests.

We can do this at the end of this file.
```diff
     },
-    scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value
+    scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value,
+    // To prevent "dead code following this construct" error when using * mockito-scala library
+    // https://github.com/mockito/mockito-scala/issues/29
+    scalacOptions in Test -= "-Ywarn-dead-code"
   )
```